### ISSUE TITLE
Action in auxiliary/post modules

### ISSUE THREAD

**Issue by scmanjarrez** [DanMcInerney/pymetasploit3#27]
Hi,
First of all, thanks for your python client, it's amazing.
I have encountered a problem when trying to set ACTION option, it isn't listed on msfrpc client neither, however, if you inspect the ruby code, it can be set using the key ACTION.
If you try to do the same in python, it fails (because it isn't listed in options) during key validation.

```
def _run_auxiliary(mod, opts)
    uuid, job = Msf::Simple::Auxiliary.run_simple(mod, {
      'Action'   => opts['ACTION'],
      'RunAsJob' => true,
      'Options'  => opts
    })
    {
      "job_id" => job,
      "uuid" => uuid
    }
  end
```
> 
>  \>\> opts = {'RHOSTS' => '10.10.0.4', 'ACTION' => 'KEYS'}
> => {"RHOSTS"=>"10.10.0.4", "ACTION"=>"KEYS"}
> \>\> rpc.call('module.execute', 'auxiliary', 'scanner/ssl/openssl_heartbleed', opts)
> => {"job_id"=>3, "uuid"=>"a9otl7jb"}
> \>\> rpc.call('job.info', 3)
> => {"jid"=>3, "name"=>"Auxiliary: scanner/ssl/openssl_heartbleed", "start_time"=>1577799485, "datastore"=>{"WORKSPACE"=>nil, "VERBOSE"=>false, "RPORT"=>443, "SSL"=>false, "SSLVersion"=>"Auto", "SSLVerifyMode"=>"PEER", "SSLCipher"=>nil, "Proxies"=>nil, "CPORT"=>nil, "CHOST"=>nil, "ConnectTimeout"=>10, "TCP::max_send_size"=>0, "TCP::send_delay"=>0, "RHOSTS"=>"10.10.0.4", "THREADS"=>1, "ShowProgress"=>true, "ShowProgressPercent"=>10, "TLS_CALLBACK"=>"None", "TLS_VERSION"=>"1.0", "MAX_KEYTRIES"=>50, "STATUS_EVERY"=>5, "DUMPFILTER"=>nil, "RESPONSE_TIMEOUT"=>10, "LEAK_COUNT"=>1, "HEARTBEAT_LENGTH"=>65535, "XMPPDOMAIN"=>"localhost", "ACTION"=>"KEYS", "RHOST"=>nil}}

>  root@3d6bd6d80ad4: \~/.msf4/loot# ls
20191231133816_default_10.10.0.4_openssl.heartble_617683.txt

Edit: Looks like it's possible to get ACTIONS available for auxiliary/post modules calling rpc.info, being able to get that information from python would be great.
```
  def rpc_info(mtype, mname)
    ...
    ...
    if m.type == 'auxiliary' || m.type == 'post'
      res['actions'] = {}
      m.actions.each_index do |i|
        res['actions'][i] = m.actions[i].name
      end

      if m.default_action
        res['default_action'] = m.default_action
      end
     ...
     ...
    res
  end
```

**Comment by DanMcInerney** [comment]
Neat, this should be an easy fix, gimme a few days or next weekend and I'll
add it in.

**Comment by scmanjarrez** [comment]
Thank you!

**Comment by scmanjarrez** [comment]
Hi, did you manage to fix it? I could try to implement it if you want.

**Comment by DanMcInerney** [comment]
I definitely wouldn't mind that. I thought it'd be as easy as adding ACTIONS to some list somewhere but I'm not clear right now how to implement.

**Comment by scmanjarrez** [comment]
Hi,
I was triying to replicate your logic for "target (exploit)" with action, but I don't know how "self.targets" is created and initialized. Do you mind explaining it?  Sorry.

**Comment by scmanjarrez** [comment]
Fixed in #36
